src/config.js

```diff
@@ -471,48 +471,57 @@ export default class Config {
    * throw an error if package.json was not found
    */
 
-  async readManifest(dir: string, priorityRegistry?: RegistryNames, isRoot?: boolean = false): Promise<Manifest> {
-    const manifest = await this.maybeReadManifest(dir, priorityRegistry, isRoot);
+  readManifest(dir: string, priorityRegistry?: RegistryNames, isRoot?: boolean = false): Promise<Manifest> {
+    return this.getCache(`manifest-${dir}`, async (): Promise<Manifest> => {
```

Comment:
yes, it's because maybeReadManifest was called few times in the same process, and the getCache there memoized "nil" return value. base-fetcher/fetch method writes the manifest between these calls, so nil was returned even after manifest were written.

moving cache to readManifest fixes the issue as getCache doesn't cache rejected promises (as opposed to nil values)